packages/core/src/ui/UICanvas.ts

```diff
@@ -0,0 +1,495 @@
+    }
+    let expectX: number, expectY: number, expectZ: number;
+    switch (this._resolutionAdaptationStrategy) {
+      case ResolutionAdaptationStrategy.WidthAdaptation:
+        expectX = expectY = expectZ = curWidth / width;
+        break;
+      case ResolutionAdaptationStrategy.HeightAdaptation:
+        expectX = expectY = expectZ = curHeight / height;
+        break;
+      case ResolutionAdaptationStrategy.BothAdaptation:
+        expectX = curWidth / width;
+        expectY = curHeight / height;
+        expectZ = (expectX + expectY) / 2;
+        break;
+      case ResolutionAdaptationStrategy.ExpandAdaptation:
+        expectX = expectY = expectZ = Math.min(curWidth / width, curHeight / height);
+        break;
+      case ResolutionAdaptationStrategy.ShrinkAdaptation:
+        expectX = expectY = expectZ = Math.max(curWidth / width, curHeight / height);
+        break;
+      default:
+        break;
+    }
+    this.entity.transform.setScale(expectX, expectY, expectZ);
+    this._transform.size.set(curWidth / expectX, curHeight / expectY);
+  }
+
+  private _walk(entity: Entity, renderers: UIRenderer[], group?: CanvasGroup): void {
+    const components = entity._components;
+    const offset = renderers.length;
+    for (let i = 0, n = components.length; i < n; i++) {
+      const component = components[i];
+      if (component.enabled) {
+        switch (component._componentType) {
+          case ComponentType.CanvasGroup:
+            const alpha = group ? group.alpha : 1;
+            group = <CanvasGroup>component;
+            group._globalAlpha = group.ignoreParentGroup ? 1 : alpha * group.alpha;
+            for (let j = offset, m = renderers.length; j < m; j++) {
+              renderers[j]._setGroup(group);
+            }
+            break;
+          case ComponentType.UIRenderer:
+            const uiRenderer = <UIRenderer>component;
+            uiRenderer._canvas = this;
+            uiRenderer._setGroup(group);
+            renderers.push(uiRenderer);
+            break;
+          default:
+            break;
+        }
+      }
+    }
+    const children = entity._children;
+    for (let i = 0, n = children.length; i < n; i++) {
+      const child = children[i];
+      child.isActive && this._walk(child, renderers, group);
+    }
+  }
+
+  private _addCameraListener(camera: Camera): void {
+    camera.entity.transform._updateFlagManager.addListener(this._onCameraTransformListener);
+    camera._updateFlagManager.addListener(this._onCameraPropertyListener);
+  }
+
+  private _removeCameraListener(camera: Camera): void {
+    camera.entity.transform._updateFlagManager.removeListener(this._onCameraTransformListener);
+    camera._updateFlagManager.removeListener(this._onCameraPropertyListener);
+  }
+
+  private _onCameraPropertyListener(flag: CameraModifyFlags): void {
+    switch (flag) {
+      case CameraModifyFlags.NearPlane:
+      case CameraModifyFlags.FarPlane:
+        break;
```

Comment:
It looks like there is an empty case block in the switch statement. Is this part of the code incomplete or planned for future implementation?